p2p/libp2p/netMessenger.go

```diff
@@ -258,15 +259,22 @@ func setupExternalP2PLoggers() {
 	}
 }
 
-func createP2PPrivKey(seed string) (*libp2pCrypto.Secp256k1PrivateKey, error) {
-	randReader, err := randFactory.NewRandFactory(seed)
+func createP2PPrivKey(p2pPrivKeyBytes []byte) (libp2pCrypto.PrivKey, error) {
+	if len(p2pPrivKeyBytes) == 0 {
+		randReader := cryptoRand.Reader
+		prvKey, _ := ecdsa.GenerateKey(btcec.S256(), randReader)
+
+		return (*libp2pCrypto.Secp256k1PrivateKey)(prvKey), nil
+	}
+
+	prvKey, err := libp2pCrypto.UnmarshalSecp256k1PrivateKey(p2pPrivKeyBytes)
 	if err != nil {
 		return nil, err
 	}
 
-	prvKey, _ := ecdsa.GenerateKey(btcec.S256(), randReader)
+	log.Warn("createP2PPrivKey: using an already existing privary key for p2p signing")
```

Comment:
Let's not use WARN here, a plain INFO will do the job. Also, we might display the ID generated from the key. There is a typo `privary`

Reply:
updated the comment; since i don't have access to the peer ID here yet i did not displayed it later since it's already visible in the listening addr string